### DISCUSSION TITLE
All Saints Day in 2025 falls on Sunday

### DISCUSSION THREAD

**Discussion by pvanthony** [romcal/romcal#320] [Q&A]
**Big correction. All Souls Day 2025 falls on Sunday and it has been moved to Monday!**

All Saints Day in 2025 falls on Sunday, so it is moved to Monday, 2nd of November and All Souls Day moved to Tuesday, 3rd of November.
Checked on romcal version 3 and it does not seem correct.

Need advice if this is a bug or I missed a configuration.

**Comment by tukusejssirs** [comment]
> All Saints Day in 2025 falls on Sunday, so it is moved to Monday, 2nd of November and All Souls Day moved to Tuesday, 3rd of November. Checked on romcal version 3 and it does not seem correct.

In the GRC, there is no such rule, e.g. in Slovakia we celebrate it on [Sunday](https://lc.kbs.sk/#20251101). However, there is a similar rule to the rule described by you in [England](https://github.com/romcal/romcal/blob/dev/lib/particular-calendars/england.ts#L260:L271).

In order to fix this issue, I need to know in which calendar(s) should the change be made. In `singapore.ts`?

**Comment by pvanthony** [comment]
Big correction. All Souls Day 2025 falls on Sunday and it has been moved to Monday.

**Reply by tukusejssirs** [reply]
You mean like this?

![Screenshot_20211103-061754_Vivaldi Snapshot](https://user-images.githubusercontent.com/5111894/140013024-9e9ea3c9-bfb8-49ae-97a2-f96d280e4fce.jpg)

For England, this is correct. For what calendar does this happen for you?

**Reply by pvanthony** [reply]
You are correct. England calendar is correct. 
We are using the following setting for the calendar.
```
const romcal = new Romcal({
    localizedCalendar: GeneralRoman_En,
    scope: 'gregorian',
    epiphanyOnSunday: true,
    corpusChristiOnSunday: true,
    ascensionOnSunday: false,
    outputOptionalMemorials: false,
    strictMode: true
});
```

**Reply by tukusejssirs** [reply]
@pvanthony, I cannot reproduce this with the latest `dev` commit (1327490); see below a sample code using manually built romcal from source.

<details>
  <summary><code>index.html</code></summary>

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <title>My Website</title>

    <script src="iife/romcal.js"></script>
    <script src="bundles/general-roman/iife/en.js"></script>
  </head>
  <body>
    <h1>romcal v3</h1>
    <hr />
    <p>Open the developer tools and check the console.</p>
    <script>
      const romcal = new Romcal({
        localizedCalendar: GeneralRoman_En,
        scope: 'gregorian',
        epiphanyOnSunday: true,
        corpusChristiOnSunday: true,
        ascensionOnSunday: false,
        outputOptionalMemorials: false,
        strictMode: true
      });

      romcal.generateCalendar(2025).then(result => {
        console.log('1 Nov 2025 :', result['2025-11-01'])
        console.log('2 Nov 2025 :', result['2025-11-02'])
      })

      // Simplified web console output
      // 1 Nov 2025 :
      //   0: Object { key: "all_saints", date: "2025-11-01", precedence: "GENERAL_SOLEMNITY_3", … }

      // 2 Nov 2025 :
      //   0: Object { key: "all_souls", date: "2025-11-02", precedence: "GENERAL_SOLEMNITY_3", … }
    </script>
  </body>
</html>
```

</details>

**Reply by tukusejssirs** [reply]
@pvanthony, you can also use `result['2025-11-01'][0].calendar.dayOfWeek)` (change the date and the index as required) to confirm the day of week. It has assigned an integer value from range `0..6`, `0` being `Sunday`. Therefore it has a value of `6` (`Saturday`) for 1 Nov 2025 and `0` (`Sunday`) for 2 Nov 2025.

**Reply by pvanthony** [reply]
I will study the England calendar and modify for Singapore because it is showing the correct values.

Thank you again for helping. I do appreciate it very much.

**Reply by tukusejssirs** [reply]
@pvanthony, just a quick question: why do you start from another particular calendar rathwn than starting from blank and add the celebrations you need?

**Reply by pvanthony** [reply]
> @pvanthony, just a quick question: why do you start from another particular calendar rathwn than starting from blank and add the celebrations you need?

Because I do not know how to make the 2025 All Saints day and All Soul day. Since England got it correct, then I can learn from their calendar and create the Singapore calendar.

**Reply by tukusejssirs** [reply]
If I were you, I would simple copy `england.ts` to `singapore.ts`, remove all celebration definitions, update the calendar configuration (see [`slovakia.ts` (L12-L16)](https://github.com/romcal/romcal/blob/dev/lib/particular-calendars/slovakia.ts#L12-L16) and finally add the celebration definitions that changed in comparison with the GRC.

Also take into account any and all celebrations celebrated on a continental level (Asia), like [St France Xavier](https://en.wikipedia.org/wiki/Patron_saints_of_places); these saints should be put into `asia.ts` calendar in romcal and they will be removed from calendars of individual Asian countries (they will inherit from `asia.ts` all celebrations and modify any celebration they need).

If you really need to modify the celebration date of All Saints and/or All Souls _in comparison with the GRC_, then you should copy that celebration from `england.ts` to `singapore.ts` and modify anything you need. However, as I understand it, in Singapore, you celebrate these two celebrations the same way as they are in the [GRC](https://github.com/romcal/romcal/blob/dev/lib/general-calendar/proper-of-saints.ts#L1054-L1066), therefore you don’t need to (re-) define them in `singapore.ts` at all. I want to emphasise this: all particular calendars inherit all celebrations from the GRC (and from continental calendars for national calendars; and from national calendars for diocesan calendars) and only the differences should be defined in the particular calendars (additions, removals, changes).

**Reply by pvanthony** [reply]
For reference this is the singapore.ts used. Currently we are testing. Friar Julian will confirm. Will update you all.
It was built by looking at england.ts and india.ts.

```
import { CalendarDef } from '../models/calendar-def';
import { Inputs } from '../types/calendar-def';

export class Singapore extends CalendarDef {
  inputs: Inputs = {
    // In England and Wales when All Saints (1 November) falls on a Saturday
    // and is transferred to Sunday, All Souls is transferred to Monday 3 November.
    // Like Ash Wednesday, All Souls is, technically, without rank.
    // However, in countries (not England & Wales) where it falls
    // on a Sunday it replaces the Sunday.
    all_saints: {
      dateDef: { month: 11, date: 1 },
      dateExceptions: {
        ifIsDayOfWeek: 6, // if is a Saturday
        setDate: { addDay: 1 },
      },
    },

    all_souls: {
      dateDef: { month: 11, date: 2 },
      dateExceptions: {
        ifIsDayOfWeek: 0, // if is a Sunday
        setDate: { addDay: 1 },
      },
    },
  };
}
```

**Reply by tukusejssirs** [reply]
@pvanthony, do I get it right that there no other changes in comparison with the GRC besides the date calculation of All Saints and All Souls?

By changes I mean _differences_ or _deviations_ from the celebration definitions in the GRC, like:
- additional celebrations;
- removed/dropped celebrations;
- changes in
  - rank (`precedence`);
  - date;
  - (potentially) colours.

**Reply by pvanthony** [reply]
Made some changes to singapore.ts and here is the final one we are using currently.
```
import { CalendarDef } from '../models/calendar-def';
import { Inputs } from '../types/calendar-def';

export class Singapore extends CalendarDef {
  inputs: Inputs = {
    // All Souls is transferred to Monday 3 November.
    // Like Ash Wednesday, All Souls is, technically, without rank.
    // This was taken from england.ts
    all_souls: {
      dateDef: { month: 11, date: 2 },
      dateExceptions: {
        ifIsDayOfWeek: 0, // if is a Sunday
        setDate: { addDay: 1 },
      },
    },
  };
}
```
As to additional, removed, we have not done anything yet to the singapore.ts
We have done some things in our javascript to get the correct readings. Here are the sections that we have done.
```
const properReadingsForMemorials = [
    'martha_of_bethany_mary_of_bethany_and_lazarus_of_bethany',
    'our_lady_of_sorrows',
    'therese_of_the_child_jesus_and_the_holy_face_of_lisieux_virgin',
    'guardian_angels',
    'barnabas_apostle',
    'passion_of_saint_john_the_baptist',
    'immaculate_heart_of_mary',
    'timothy_of_ephesus_and_titus_of_crete_bishops',
    'mary_mother_of_the_church',
  ];


                    else if (result[0].rank == "MEMORIAL" && properReadingsForMemorials.includes(result[0].key)){
                        var path = baseURL + "/" + result[0].rank + "/" + result[0].key;
                        document.getElementById("liturgicalReading").innerHTML = result[0].name + " (" + result[0].cycles.weekdayCycle + ")";
                        document.getElementById("liturgicalDay").innerHTML = "";
                    }
    
                    else if (result[0].rank == "MEMORIAL"){
                        var path = baseURL + result[0].seasons[0] + "/" + "WEEKDAY" + "/" + result[0].cycles.weekdayCycle  + "/" + String(result[0].calendar.weekOfSeason).padStart(2, '0') + "/" + result[0].calendar.dayOfWeek;
                        document.getElementById("liturgicalReading").innerHTML = result[0].weekday.name + " (" + result[0].cycles.weekdayCycle + ")";
                        document.getElementById("liturgicalDay").innerHTML = result[0].name;
                    }
```

There is one more place we did a change to get the correct readings.
```
                    else if (result[0].periods[0]== "DAYS_BEFORE_EPIPHANY"){
                        if (result[0].key == "basil_the_great_and_gregory_nazianzen_bishops"){
                            var path = baseURL + "CHRISTMASTIDE" + "/" + "WEEKDAY" + "/" + result[0].cycles.weekdayCycle + "/EPIPHANY/" + result[0].weekday.key
                            document.getElementById("liturgicalReading").innerHTML = result[0].weekday.name + " (" + result[0].cycles.weekdayCycle + ")";
                            document.getElementById("liturgicalDay").innerHTML = "";
                        }
                        else {
                            var path = baseURL + "CHRISTMASTIDE" + "/" + "WEEKDAY" + "/" + result[0].cycles.weekdayCycle + "/EPIPHANY/" + result[0].key
                            document.getElementById("liturgicalReading").innerHTML = result[0].name + " (" + result[0].cycles.weekdayCycle + ")";
                            document.getElementById("liturgicalDay").innerHTML = "";
                        }
                    }
```
Other then that we have not done much changes.

Here is the link to the english version of the site.
https://missal.org/

Here is the link to our javascript used on missal.org
https://missal.org/packages/julian_custom.js

Hope I have answered the questions.
